contracts/polyx-vesting/lib.rs

```diff
@@ -0,0 +1,253 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract]
+mod polyx_vesting {
+    use ink_storage::traits::SpreadAllocate;
+
+    /// Defines the storage of your contract.
+    #[ink(storage)]
+    #[derive(SpreadAllocate)]
+    pub struct PolyxVesting {
+        released: Balance,
+        beneficiary: AccountId,
+        start: Timestamp,
+        duration: Timestamp,
+    }
+
+    /// Event emitted when Polyx is released.
+    #[ink(event)]
+    pub struct PolyxReleased {
+        value: Balance,
+    }
+
+    /// The error types.
+    #[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub enum Error {
+        /// Returned if not enough balance to fulfill a request is available.
+        InsufficientBalance,
+        /// Funds are not released as yet.
+        FundsNotReleased,
+        /// Invalid start timestamp
+        InvalidStartTimestamp,
+        /// Duration Overflow
+        DurationOverflow,
+        /// Invalid timestamp
+        InvalidTimestamp,
+        /// Timestamp Overflow
+        TimestampOverflow,
+    }
+
+    /// The contract result type.
+    pub type Result<T> = core::result::Result<T, Error>;
+
+    impl PolyxVesting {
+        /// Constructor
+        #[ink(constructor, payable)]
+        pub fn new(
+            beneficiary_address: AccountId,
+            start_timestamp: Timestamp,
+            duration_milli_seconds: Timestamp,
+        ) -> Result<Self> {
+            ink_lang::utils::initialize_contract(|contract| {
+                Self::new_init(
+                    contract,
+                    beneficiary_address,
+                    start_timestamp,
+                    duration_milli_seconds,
+                )
+            })
+        }
+
+        fn new_init(
+            &mut self,
+            beneficiary_address: AccountId,
+            start_timestamp: Timestamp,
+            duration_milli_seconds: Timestamp,
+        ) -> Result<()> {
+            self.beneficiary = beneficiary_address;
+            self.start = start_timestamp;
+            self.duration = duration_milli_seconds;
+            self.validation()
+        }
+
+        fn validation(&self) -> Result<()> {
+            // Ensure start isn't zero
+            if self.start == 0 || self.duration == 0 {
+                return Err(Error::InvalidTimestamp);
+            }
+
+            let max_milli_seconds = u64::MAX;
+            // Ensure start and duration not over type limit
+            if self.start < ink_env::block_timestamp::<ink_env::DefaultEnvironment>() {
+                return Err(Error::InvalidStartTimestamp);
+            } else if self.duration > max_milli_seconds {
+                return Err(Error::DurationOverflow);
```

Comment:
The `Timestamp` type is `u64` , so `duration` can't be greater  than `u64::MAX`.  This check will never fail.

Please add a check that `start + duration` doesn't overflow.